src/util/alloc/free_list_allocator.rs

```diff
@@ -0,0 +1,530 @@
+            &mut self.available_blocks_stress,
+            &mut self.consumed_blocks,
+            size,
+            align,
+        );
+        if block.is_zero() {
+            block = self.recycle_local_blocks(size, align, true);
+        }
+        if block.is_zero() {
+            block = self.acquire_fresh_block(size, align, true);
+        }
+        block
+    }
+
+    // Find an available block from local block lists
+    #[inline(always)]
+    fn find_free_block_local(&mut self, size: usize, align: usize) -> Block {
+        let block = Self::find_free_block_with(
+            &mut self.available_blocks,
+            &mut self.consumed_blocks,
+            size,
+            align,
+        );
+        if block.is_zero() {
+            self.recycle_local_blocks(size, align, false)
+        } else {
+            block
+        }
+    }
+
+    // Find an available block
+    // This will usually be the first block on the available list. If all available blocks are found
+    // to be full, other lists are searched
+    // This function allows different available block lists -- normal allocation uses self.avaialble_blocks, and precise stress test uses self.avialable_blocks_stress.
+    #[inline(always)]
+    fn find_free_block_with(
+        available_blocks: &mut BlockLists,
+        consumed_blocks: &mut BlockLists,
+        size: usize,
+        align: usize,
+    ) -> Block {
+        let bin = mi_bin::<VM>(size, align);
+        debug_assert!(bin <= MAX_BIN);
+
+        let available = &mut available_blocks[bin];
+        debug_assert!(available.size >= size);
+
+        if !available.is_empty() {
+            let mut block = available.first;
+
+            while !block.is_zero() {
+                if block.has_free_cells() {
+                    return block;
+                }
+                available.pop();
+                consumed_blocks.get_mut(bin).unwrap().push(block);
+
+                block = available.first;
+            }
+        }
+
+        debug_assert!(available_blocks[bin].is_empty());
+        Block::ZERO_BLOCK
+    }
+
+    /// Add a block to the given bin in the available block lists. Depending on which available block list we are using, this
+    /// method may add the block to available_blocks, or available_blocks_stress.
+    #[inline(always)]
+    fn add_to_available_blocks(&mut self, bin: usize, block: Block, stress: bool) {
+        if stress {
+            debug_assert!(self.plan.base().is_precise_stress());
+            self.available_blocks_stress[bin].push(block);
+        } else {
+            self.available_blocks[bin].push(block);
+        }
+    }
+
+    /// Tries to recycle local blocks if there is any. This is a no-op for eager sweeping mark sweep.
+    #[cfg(not(feature = "eager_sweeping"))]
+    #[inline]
+    fn recycle_local_blocks(&mut self, size: usize, align: usize, _stress_test: bool) -> Block {
+        // attempt to sweep
+        loop {
```

Comment:
We could just move the `#[cfg(...)]` macro inside the function.
```suggestion
    #[inline]
    fn recycle_local_blocks(&mut self, size: usize, align: usize, _stress_test: bool) -> Block {
        if cfg!(not(feature = "eager_sweeping")) {
            return Block::ZERO_BLOCK; // We should really make Block non-zero and return None here.
        }    
        // attempt to sweep
        loop {
```

Reply:
Done